src/UnitTests/IMappingExpression/ForCtorParam.cs

```diff
@@ -51,5 +55,52 @@ public void Should_resolve_using_custom_func()
 
             dest.Value1.ShouldEqual(8);
         }
+
+        [Fact]
+        public void Should_ignore_nonexistent_parameter()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>()
+                    .ForCtorParam("thing", opt => opt.MapFrom(src => src.Value))
+                    .ForCtorParam("think", opt => opt.MapFrom(src => src.Value));
+            });
+
+            config.AssertConfigurationIsValid();
+
+            var result = config.CreateMapper().Map<Dest>(new Source { Value = 42 });
+
+            result.Value1.ShouldEqual(42);
+        }
+
+        [Fact]
+        public void Should_ignore_when_no_constructor_is_present()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, DestWithNoConstructor>()
+                    .ForMember(dest => dest.Value1, opt => opt.MapFrom(src => src.Value))
+                    .ForCtorParam("thing", opt => opt.MapFrom(src => src.Value));
+            });
+
+            config.AssertConfigurationIsValid();
+
+            var result = config.CreateMapper().Map<DestWithNoConstructor>(new Source { Value = 17 });
+
+            result.Value1.ShouldEqual(17);
+        }
+
+        [Fact]
+        public void Should_not_pass_config_validation_when_parameter_is_mispelt()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>()
+                    .ForCtorParam("think", opt => opt.MapFrom(src => src.Value));
+            });
+
+            Action configValidation = () => config.AssertConfigurationIsValid();
+            configValidation.ShouldThrow<AutoMapperConfigurationException>();
```

Comment:
When the mapping is called, an `ArgumentException` is thrown with the message "Dest needs to have a constructor with 0 args or only optional args", i.e. the same behavior as if the setup had been just `cfg.CreateMap<Source, Dest>()`.